pkg/services/ngalert/accesscontrol/silences.go

```diff
@@ -0,0 +1,252 @@
+package accesscontrol
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	ac "github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+)
+
+const (
+	instancesRead   = ac.ActionAlertingInstanceRead
+	instancesCreate = ac.ActionAlertingInstanceCreate
+	instancesWrite  = ac.ActionAlertingInstanceUpdate
+	silenceRead     = ac.ActionAlertingSilencesRead
+	silenceCreate   = ac.ActionAlertingSilencesCreate
+	silenceWrite    = ac.ActionAlertingSilencesWrite
+)
+
+var (
+	// asserts full read-only access to silences
+	readAnySilenceEvaluator = ac.EvalPermission(instancesRead)
+	// asserts that user has access to silences at all
+	readSomeSilenceEvaluator = ac.EvalAny(ac.EvalPermission(instancesRead), ac.EvalPermission(silenceRead))
+	// asserts whether user has read access to rules in a specific folder
+	readRuleSilenceEvaluator = func(folderUID string) ac.Evaluator {
+		return ac.EvalAny(
+			ac.EvalPermission(instancesRead),
+			ac.EvalPermission(silenceRead, dashboards.ScopeFoldersProvider.GetResourceScopeUID(folderUID)),
+		)
+	}
+
+	// asserts that user has access to create any silence (ie full create access)
+	createAnySilenceEvaluator = ac.EvalAll(ac.EvalPermission(instancesCreate), readAnySilenceEvaluator)
+	// asserts that user has access to create global silences
+	createGlobalSilenceEvaluator = ac.EvalAll(ac.EvalPermission(instancesCreate), readSomeSilenceEvaluator)
```

Comment:
What is the intended difference here? Is the idea that a user could create global silences but not rule-based silences?

createAnySilenceEvaluator:
```
instance:create + instance:read
```

createGlobalSilenceEvaluator:
```
instance:create + any(instance:read, silence:read)
```